### ISSUE TITLE
Easy update process

### ISSUE THREAD

**Issue by konceptech** [jekkos/opensourcepos#736]
Is there a way to implement an easy update process, trough the UI or maybe trough an php wizard.

This is mainly for setup that are hosted on shared environement, with no access to the console.

**Comment by jekkos** [comment]
Something could be set up using git and built in CI db migration functionality

**Comment by Modding** [comment]
you can also create a repository like on Wordpress.

**Comment by daN4cat** [comment]
Probably this is something we could reuse: https://github.com/jimdoescode/CodeIgniter-GitHub-Updater

It would be indeed nice to have an updating feature pulling the code from github. When back from holidays I will have a look into that lib, for sure it needs an update to CI3.

**Comment by daN4cat** [comment]
However one problem to solve would be any database upgrade script, but I guess something can be sorted.

**Comment by jekkos** [comment]
For database CI_Migration could be an option. But then I guess we will only support this from 3.0+ onwards (otherwise we'll need to migrate all existing upgrade scripts to this format).

Added advantage is that mysql will be used through the CI connection and thus no extra client needs to be configured.

**Comment by daN4cat** [comment]
Only after 3.0.0, not now.

**Comment by daN4cat** [comment]
CI Migration is supported, maybe one day we can add the automatic download of the code, backup of the old code and database and upgrade of the code.
